### ISSUE TITLE
indent rule can break on arrayExpression inside a CallExpression

### ISSUE THREAD

**Issue by Trott** [eslint/eslint#7747]
* **ESLint Version:** 3.12.0
* **Node Version:** 7.2.1
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
no configuration

```




```

**What did you do? Please include the actual source code causing the issue.**

Running this:

```sh
eslint --no-eslintrc --rule='indent: [2, 2, {CallExpression: {arguments: first}, ArrayExpression: first}]' test.js
```

on file `test.js`:

```js
echo = spawn('cmd.exe',
             ['/c', 'echo', 'hello&&', 'echo',
              'node&&', 'echo', 'and&&', 'echo', 'world']);





```

**What did you expect to happen?**

No error

**What actually happened? Please include the actual, raw output from ESLint.**

```console
/Users/trott/test/test.js
  2:14  error  Expected indentation of 2 spaces but found 13  indent

✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Thanks, I can reproduce this issue.

**Comment by Trott** [comment]
I think this was closed in error. This problem still exists in 3.12.2.

**Comment by Trott** [comment]
Er, wait, no it doesn't...never mind me.
